cmd/kots/cli/velero.go

```diff
@@ -63,6 +83,299 @@ func EnsurePermissionsCmd() *cobra.Command {
+		Long:          ``,
+		SilenceUsage:  true,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			v := viper.GetViper()
+
+			namespace := v.GetString("namespace")
+			if err := validateNamespace(namespace); err != nil {
+				return err
+			}
+
+			hostPath := v.GetString("hostpath")
+
+			if hostPath == "" {
+				return errors.New("--hostpath option is required")
+			}
+
+			fileSystemConfig := snapshottypes.FileSystemConfig{
+				HostPath: &hostPath,
+			}
+
+			registryOptions, err := getRegistryConfig(v)
+			if err != nil {
+				return errors.Wrap(err, "failed to get registry config")
+			}
+
+			log := logger.NewCLILogger()
+
+			return veleroConfigureFileSystem(cmd.Context(), namespace, log, fileSystemConfig, registryOptions, v.GetBool("skip-validation"), v.GetString("output"))
+		},
+	}
+
+	cmd.Flags().String("hostpath", "", "a local host path on the node")
+	cmd.Flags().StringP("namespace", "n", "", "the namespace in which kots/kotsadm is installed")
+	cmd.Flags().Bool("skip-validation", false, "skip the validation of the store configuration")
+	cmd.Flags().StringP("output", "o", "", "output format. supported values: json")
+
+	registryFlags(cmd.Flags())
+
+	return cmd
+}
+
+func veleroConfigureFileSystem(ctx context.Context, namespace string, log *logger.CLILogger, fileSystemConfig snapshottypes.FileSystemConfig, registryOptions *kotsadmtypes.KotsadmOptions, skipValidation bool, outputFormat string) error {
+	log.ActionWithSpinner("Setting up File System Minio")
+
+	clientset, err := k8sutil.GetClientset(kubernetesConfigFlags)
+	if err != nil {
+		return errors.Wrap(err, "failed to get clientset")
+	}
+
+	deployOptions := snapshot.FileSystemDeployOptions{
+		Namespace:        namespace,
+		IsOpenShift:      k8sutil.IsOpenShift(clientset),
+		FileSystemConfig: fileSystemConfig,
+	}
+	if err := snapshot.DeployFileSystemMinio(ctx, clientset, deployOptions, *registryOptions); err != nil {
+		if _, ok := errors.Cause(err).(*snapshot.ResetFileSystemError); ok {
+			log.FinishSpinnerWithWarning(color.New(color.FgHiRed))
+			forceReset := promptForFileSystemReset(log, err.Error())
+			if forceReset {
+				log.ActionWithSpinner("Re-configuring File System Minio")
+				deployOptions.ForceReset = true
+				if err := snapshot.DeployFileSystemMinio(ctx, clientset, deployOptions, *registryOptions); err != nil {
+					log.FinishSpinnerWithError()
+					return errors.Wrap(err, "failed to force deploy file system minio")
+				}
+			}
+		} else {
+			log.FinishSpinnerWithError()
+			return errors.Wrap(err, "failed to deploy file system minio")
+		}
+	}
+
+	log.FinishSpinner()
+	log.ActionWithSpinner("Waiting for File System Minio to be ready")
+
+	err = k8sutil.WaitForDeploymentReady(ctx, clientset, namespace, snapshot.FileSystemMinioDeploymentName, time.Minute*2)
```

Comment:
TODO: allow user specify this on command line.  in other places too.

Reply:
if you're referring to the timeout period, then yeah. but the wait step is necessary in this case because we have to ensure/create the bucket after.